### PR TITLE
cabal.project: use multiple subdirs

### PR DESCRIPTION
**Pull request by mrBliss**
#2461 was closed when the `consensus-improvements-after-hardwork` branch was merged.

### CODE REVIEW

**Comment by mrBliss** [comment]
Draft because this includes #2503 to avoid conflicts when that branch is merged.

**Comment by coot** [comment]
@mrBliss which `cabal-install` version do we need? `3.4` or `3.2`?

**Comment by mrBliss** [comment]
> @mrBliss which `cabal-install` version do we need? `3.4` or `3.2`?

3.2

File: cabal.project [hunk 1]
```diff
@@ -120,7 +120,7 @@ source-repository-package
   location: https://github.com/input-output-hk/iohk-monitoring-framework
   tag: d8f35c3c725be67148681227f9d64876f5909120
   --sha256: 09v1m9km3arzxaqlyaw576w69fwfi02q7nm5ias162sfzzg8za4x
-  subdir:   contra-tracer
+  subdir: contra-tracer
 
 source-repository-package
```
Comment (coot) [inline cabal.project:125]: Could you also transform `cardano-prelude`?

Comment (mrBliss) [reply cabal.project:125]: No, see the description of the second commit:

> For some reason, using `.` as a subdir doesn't seem to be supported. For now,
> work around it by having two entries for `cardano-prelude`.

**Comment by mrBliss** [comment]
bors merge

**Comment by mrBliss** [comment]
bors merge

**Comment by mrBliss** [comment]
bors merge

**Comment by nc6** [comment]
bors merge

**Comment by mrBliss** [comment]
4 timeouts, there must be some CI issue. 

@nc6 Yet, https://github.com/input-output-hk/cardano-ledger-specs/pull/1747/ was merged. Hmm, does bors still work on `cardano-ledger-specs`? I don't see any PRs bors'ed after that one.

**Comment by mrBliss** [comment]
bors merge

**Comment by mrBliss** [comment]
I bumped the dependency on haskell.nix again, hopefully that fixes bors (there were some commits related `cabal.project` and subdir parsing).

**Comment by mrBliss** [comment]
bors merge

**Comment by nc6** [comment]
> 4 timeouts, there must be some CI issue.
> 
> @nc6 Yet, [input-output-hk/cardano-ledger-specs#1747](https://github.com/input-output-hk/cardano-ledger-specs/pull/1747) was merged. Hmm, does bors still work on `cardano-ledger-specs`? I don't see any PRs bors'ed after that one.

We very rarely use bors there, so I doubt it.

**Comment by mrBliss** [comment]
bors merge

**Pull request merged**